### DISCUSSION TITLE
Automated facial skin analysis system based on 3D scanning

### DISCUSSION THREAD

**Discussion by Fridayevery** [community/community#130950] [Programming Help]
### Body

Hey, everyone. I am going to do a project and want to use a 3D scanner for facial skin analysis. After consultation with my tutor, I chose [POP3](https://www.revopoint3d.com/products/portable-3d-scanner-pop3). The basic workflow is to collect the 3D model and texture information of the face through 3D scanning, and then import it into the development application for automatic processing and analysis, such as evaluating skin quality, skin color, spots, wrinkles, etc., and generating a visual report.

![73126c0eb41c17d1a21125b12a606544](https://github.com/community/community/assets/157455221/e84302ef-2f9c-40af-8b17-d80e9d61a638)

Therefore, I need an automated software system/application to complete this operation. But I did not search for relevant open source code information on github. Can you guide me to solve this problem? Or give me some suggestions. The system will obtain 3D models and texture information from 3D scanners and automatically analyze skin-related items.

If you have any good suggestions, I will be very grateful. Thank you in advance

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Dawoodoz** [comment]
Doing image analysis on a scanned 3D model instead of real photos is going to make calibration very difficult, because you want to see the small patterns that are not well preserved. Skin defects are also much more visible in the ultra violet spectrum than a plain diffuse texture. Skin needs to be looked at from multiple angles with different lights to get all the details.

My suggestion is to build your own scanner with flashing LEDs of different visible and invisible wavelengths. Ultra violet light shows defects as clearly visible black spots. Lights flashing from different directions will show bumps and roughness. Then you use dense optical flow to match images that are close in time using a wavelength with constant light while gathering overlapping photos with light from different directions. This will give you highly detailed bumps, wide spectrum color and specular information about microscopic patterns in a 2D format that can be fed with manual annotation to a deep convolution learning system. You might need at least 100 volunteering test subjects to get decent results without overfitting. The results would then be displayed on photos of the person from different angles and can later be stitched to a 3D model if needed.
